beacon_node/execution_layer/src/lib.rs

```diff
@@ -254,8 +303,10 @@ impl ExecutionLayer {
         timestamp: u64,
         random: Hash256,
         finalized_block_hash: Hash256,
+        proposer_index: Option<u64>,
```

Comment:
I think we can just use a `u64` here, rather than an `Option<u64>`. It would make the function less flexible, but I think the simplicity is worth it.

Reply:
I ended up addressing this in https://github.com/paulhauner/lighthouse/commit/07a6a43eaf06d59334177c82293b73ce9f7e95a8.

Reply:
Used an `Option<u64>` there to be able to pass `None` in the [test code](https://github.com/sigp/lighthouse/blob/07a6a43eaf06d59334177c82293b73ce9f7e95a8/beacon_node/execution_layer/src/test_utils/mock_execution_layer.rs#L130-L139) that used that function 😉
But just passing 0 as `proposer_index` in the test seems fine, too and makes the Option unnecessary 👍